Expected error status if search with partition_tag not existed

**Describe the bug**
Current result is `code=0, message='Search vectors successfully!'`
if partition_tag has been not created

```
________________________________________________________________________________________________________________________________________________ TestSearchParamsInvalid.test_search_with_tag_not_existed ________________________________________________________________________________________________________________________________________________

self = <test_search_vectors.TestSearchParamsInvalid object at 0x7fc299c8ef28>, connect = <milvus.client.stub.Milvus object at 0x7fc299c8e908>, collection = 'test_search_xpOSoqem'

    @pytest.mark.level(1)
    def test_search_with_tag_not_existed(self, connect, collection):
        top_k = 1
        nprobe = 1
        query_vecs = gen_vectors(1, dim)
        status, result = connect.search_vectors(collection, top_k, query_vecs, partition_tags=["tag"])
        logging.getLogger().info(result)
>       assert not status.OK()
E       AssertionError: assert not True
E        +  where True = <bound method Status.OK of Status(code=0, message='Search vectors successfully!')>()
E        +    where <bound method Status.OK of Status(code=0, message='Search vectors successfully!')> = Status(code=0, message='Search vectors successfully!').OK

test_search_vectors.py:1060: AssertionError
------------------------------------------------------------------------------------------------------------------------------------------------------------------- Captured log setup -------------------------------------------------------------------------------------------------------------------------------------------------------------------
[14:10:44-INFO-root]: Status(code=0, message='Success') (conftest.py:45)
------------------------------------------------------------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------------------------------------------------------------
[14:10:44-INFO-root]: [

] (test_search_vectors.py:1059)
```

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation (Docker, or from source)
- Milvus version (v0.3.1, or v0.4.0)
- Milvus configuration (Settings you made in `server_config.yaml`)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,6 +18,7 @@ Please mark all change in change log and use the issue from GitHub
 -   \#1724 Remove unused unittests
 -   \#1734 Opentracing for combined search request
 -   \#1735 Fix search out of memory with ivf_flat 
+-   \#1747 Expected error status if search with partition_tag not existed
 
 ## Feature
 -   \#1603 BinaryFlat add 2 Metric: Substructure and Superstructure

--- a/core/src/db/DBImpl.cpp
+++ b/core/src/db/DBImpl.cpp
@@ -1132,7 +1132,7 @@ DBImpl::Query(const std::shared_ptr<server::Context>& context, const std::string
         std::set<std::string> partition_name_array;
         status = GetPartitionsByTags(table_id, partition_tags, partition_name_array);
         if (!status.ok()) {
-            return status; // didn't match any partition.
+            return status;  // didn't match any partition.
         }
 
         for (auto& partition_name : partition_name_array) {

--- a/core/src/db/DBImpl.cpp
+++ b/core/src/db/DBImpl.cpp
@@ -1130,14 +1130,17 @@ DBImpl::Query(const std::shared_ptr<server::Context>& context, const std::string
     } else {
         // get files from specified partitions
         std::set<std::string> partition_name_array;
-        GetPartitionsByTags(table_id, partition_tags, partition_name_array);
+        status = GetPartitionsByTags(table_id, partition_tags, partition_name_array);
+        if (!status.ok()) {
+            return status; // didn't match any partition.
+        }
 
         for (auto& partition_name : partition_name_array) {
             status = GetFilesToSearch(partition_name, files_array);
         }
 
         if (files_array.empty()) {
-            return Status(PARTITION_NOT_FOUND, "Cannot find the specified partitions");
+            return Status::OK();
         }
     }
 
@@ -1663,6 +1666,10 @@ DBImpl::GetPartitionsByTags(const std::string& table_id, const std::vector<std::
                 partition_name_array.insert(schema.table_id_);
             }
         }
+    }
+
+    if (partition_name_array.empty()) {
+        return Status(PARTITION_NOT_FOUND, "Cannot find the specified partitions");
     }
 
     return Status::OK();

--- a/core/src/db/DBImpl.cpp
+++ b/core/src/db/DBImpl.cpp
@@ -1137,7 +1137,7 @@ DBImpl::Query(const std::shared_ptr<server::Context>& context, const std::string
         }
 
         if (files_array.empty()) {
-            return Status::OK();
+            return Status(PARTITION_NOT_FOUND, "Cannot find the specified partitions");
         }
     }
 

--- a/core/src/utils/Error.h
+++ b/core/src/utils/Error.h
@@ -93,6 +93,7 @@ constexpr ErrorCode DB_INCOMPATIB_META = ToDbErrorCode(6);
 constexpr ErrorCode DB_INVALID_META_URI = ToDbErrorCode(7);
 constexpr ErrorCode DB_EMPTY_TABLE = ToDbErrorCode(8);
 constexpr ErrorCode DB_BLOOM_FILTER_ERROR = ToDbErrorCode(9);
+constexpr ErrorCode PARTITION_NOT_FOUND = ToDbErrorCode(10);
 
 // knowhere error code
 constexpr ErrorCode KNOWHERE_ERROR = ToKnowhereErrorCode(1);

--- a/tests/milvus_python_test/test_search_vectors.py
+++ b/tests/milvus_python_test/test_search_vectors.py
@@ -327,8 +327,7 @@ class TestSearchBase:
         search_param = get_search_param(index_type)
         status, result = connect.search_vectors(collection, top_k, query_vec, partition_tags=["new_tag"], params=search_param)
         logging.getLogger().info(result)
-        assert status.OK()
-        assert len(result) == 0
+        assert not status.OK()
 
     def test_search_l2_index_params_partition_E(self, connect, collection, get_simple_index):
         '''

